pkg/generic/map_test/generic_test.go

```diff
@@ -42,11 +42,171 @@ func TestThrift(t *testing.T) {
 
 	cli := initThriftClient(t, "127.0.0.1:9021")
 
-	resp, err := cli.GenericCall(context.Background(), "ExampleMethod", reqMsg, callopt.WithRPCTimeout(100*time.Second))
-	test.Assert(t, err == nil, err)
-	respMap, ok := resp.(map[string]interface{})
-	test.Assert(t, ok)
-	test.Assert(t, reflect.DeepEqual(respMap, respMsg), respMap)
+	cases := []struct {
+		reqMsg   interface{}
+		wantResp interface{}
+	}{
+		{
+			reqMsg: map[string]interface{}{
+				"Msg": "hello",
+				"TestList": []interface{}{
+					map[string]interface{}{
+						"Bar": "foo",
+					},
+				},
+				"TestMap": map[interface{}]interface{}{
+					"l1": map[string]interface{}{
+						"Bar": "foo",
+					},
+				},
+				"StrList": []interface{}{
+					"123", "456",
+				},
+				"I64List": []interface{}{
+					int64(123), int64(456),
+				},
+				"B": true,
+			},
+			wantResp: map[string]interface{}{
+				"Msg": "hello",
+				"TestList": []interface{}{
+					map[string]interface{}{
+						"Bar": "foo",
+					},
+				},
+				"TestMap": map[interface{}]interface{}{
+					"l1": map[string]interface{}{
+						"Bar": "foo",
+					},
+				},
+				"StrList": []interface{}{
+					"123", "456",
+				},
+				"I64List": []interface{}{
+					int64(123), int64(456),
+				},
+				"B": true,
+			},
+		},
+		{
+			reqMsg: map[string]interface{}{
+				"Msg": "hello",
+				"TestList": []interface{}{
+					map[string]interface{}{
+						"Bar": "foo",
+					},
+					nil,
+				},
+				"TestMap": map[interface{}]interface{}{
+					"l1": map[string]interface{}{
+						"Bar": "foo",
+					},
+					"l2": nil,
+				},
+				"StrList": []interface{}{
+					"123", nil,
+				},
+				"I64List": []interface{}{
+					int64(123), nil,
+				},
+				"B": nil,
+			},
+			wantResp: map[string]interface{}{
+				"Msg": "hello",
+				"TestList": []interface{}{
+					map[string]interface{}{
+						"Bar": "foo",
+					},
+					map[string]interface{}{},
+				},
+				"TestMap": map[interface{}]interface{}{
+					"l1": map[string]interface{}{
+						"Bar": "foo",
+					},
+					"l2": map[string]interface{}{},
+				},
+				"StrList": []interface{}{
+					"123", "",
+				},
+				"I64List": []interface{}{
+					int64(123), int64(0),
+				},
+			},
+		},
+		{
+			reqMsg: map[string]interface{}{
+				"Msg": "hello",
+				"TestList": []interface{}{
+					nil,
+				},
+				"TestMap": map[interface{}]interface{}{
+					"l2": nil,
+				},
+				"StrList": []interface{}{
+					nil,
+				},
+				"I64List": []interface{}{
+					nil,
+				},
+				"B": nil,
+			},
+			wantResp: map[string]interface{}{
+				"Msg": "hello",
+				"TestList": []interface{}{
+					map[string]interface{}{},
+				},
+				"TestMap": map[interface{}]interface{}{
+					"l2": map[string]interface{}{},
+				},
+				"StrList": []interface{}{
+					"",
+				},
+				"I64List": []interface{}{
+					int64(0),
+				},
+			},
+		},
+		{
```

Comment:
duplicated test cases?

Reply:
I just wanted to test writeStringMap (not writeInterfaceMap) for an experiment. So actually this is not duplicated one but I'll delete this.

Reply:
Ohhh I see.